src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/PackageRestoreBar.xaml.cs

```diff
@@ -99,9 +113,20 @@ private void UserControl_Loaded(object sender, RoutedEventArgs e)
                     try
                     {
                         string solutionDirectory = await _solutionManager.GetSolutionDirectoryAsync(CancellationToken.None);
+                        _componentModel = await AsyncServiceProvider.GlobalProvider.GetComponentModelAsync();
+                        _vsSolutionManager = _componentModel?.GetService<IVsSolutionManager>();
 
                         // when the control is first loaded, check for missing packages
-                        await _packageRestoreManager.RaisePackagesMissingEventForSolutionAsync(solutionDirectory, CancellationToken.None);
+                        if (_projectContextInfo?.ProjectStyle == ProjectModel.ProjectStyle.PackageReference &&
+                            await GetMissingAssetsFileStatusAsync(_projectContextInfo.ProjectId))
+                        {
+                            _solutionRestoreWorker = _componentModel?.GetService<ISolutionRestoreWorker>();
```

Comment:
If component model can be null, then that'd mean solutioonrestoreworker could be null. 

Basically it feels like if something fails we end up with an NRE. 

Consider cleaning that up. Most of our code doesn't bother checking componentModel for null, as everything would be failing anyways if that's not true.